protected/humhub/components/ApplicationTrait.php

```diff
@@ -27,6 +29,12 @@ trait ApplicationTrait
      */
     public $minSupportedPhpVersion;
 
+    /**
+     * @var MigrationInterface|null Will be set to the instance of the currently running migration
+     * @see Migration::up() and Migration::down()
+     */
+    public ?MigrationInterface $currentMigration = null;
```

Comment:
What do you think about using the new `MigrationService` for this?

```
public static ?MigrateInterface $currentMigration = null;
```

Reply:
That's an interesting idea.

- pro:
  - In terms of logic (as in what belongs where) I love the idea
- contra
  - in terms of efficiency: the `MigrationService` class would have to be loaded not only during migration, but also during regular request cycles, as the _consuming_ automation would load the class just to find the field empty. The application object is already loaded ... (for the same reason I've not stored it as a "component", but directly as a fixed field).

Happy to adjust to your verdict ... :-)

Reply:
> * the _consuming_ automation would load the class just to find the field empty

Well, we could do a 
```php
if (class_exists(humhub\services\MigrationService::$class, false) && MigrationService::$currentMigration !== null) {
   // do what needs to be done during migration
} else {
   // to the regular stuff
}
```

Reply:
@martin-rueegg Can you give an example of when we need to use this feature?

Do I understand correctly that some events have to check whether migrations are currently being executed?

Reply:
If had similar situations before.

Currently I use it here, where I would update a polymorphic relation to synchronize `guid` and `gid`. But this can only happen _after_ the migration is applied. Yet, I use those object so migrate the data. An alternative approach would be to create a mock object during migration ...

https://github.com/humhub/humhub/pull/6718/files#diff-86ef7d668695f939c218131eb996b6c3218e640f46349304efc41b3d51babe2fR54-R81

Reply:
Sorry, I didn't quite get it.
Normally the migrations are executed separately and the models do not have to check this?

Reply:
The thing is, that I wanted to add the `gid` field to `contentcontainer`, `space`, and `user` tables. To fill them with initial values, I have to generate the GlobalId record of which I then read the `gid` and save it to the respective table.

But when if I do one table at the time (since their migrations are in different modules) some fields do not yet exist, that are updated from the `GlobalId::afterSave()` method.

But I can have a look to see if there's another way to solve this.

Reply:
I hadn't time yet, to take a closer look at the Global ID PR and concept. 

But in principle, a migration for core models would be like this.

// Example Migration
``` 
function migrateUp() {
    foreach (GetRowsInUserTable() as $row) {
               $id = InsertRowInGidTable();
               UpdateUserRowWithGid();
     }
}
```

 It would be good if DB migrations work as low-level as possible without Models (afterSave etc.), as otherwise Model changes may break future migrations.

Reply:
Yes, that makes sense. Thank you.

so the `GetRowsInUserTable();` and `InsertRowInGidTable();` would be something to include in the migration class, right?

Reply:
I would usually use code like this in the migration: 

```php
    function up()
    {
        foreach ((new \yii\db\Query())->select("*")->from('user')->all() as $row) {
            $this->insert('gid', [/* DATA FOR GID Table */]);
            $this->update('user', ['gid' => $db->lastInsertId()], ['id' => $row['id']]);
        }
    }
```

In case for global id and also the need to migrate lot of module data in future, I would add a method like this:

// humhub\components\Migration
```php
   function createGidRecord($class, $id): int {
            $this->insert('gid', [/* DATA FOR GID Table */]);
           return $db->lastInsertId();
   }
``` 
 
// in migration e.g. for Users 
```php
    function up()
    {
        foreach ((new \yii\db\Query())->select("*")->from('user')->all() as $row) {
            $this->update('user', ['gid' => $this->createGidRecord(User::class, $row['id'])], ['id' => $row['id']]);
        }
    }
```

Reply:
Thank you for your extended example. Much appreciated!